### DISCUSSION TITLE
Share state in Chrome Extension Service Worker and PopUp view

### DISCUSSION THREAD

**Discussion by mzvdev** [pmndrs/zustand#2020] [Q&A]
Hi,

I'm building a chrome extension where I want to share state between  service worker and the popup view.
I'm able to persist state in the Vanilla Zustand store, but when I update the store in the service worker, the changes don't show up in the PopUp view until I close then open the PopUp.

I can subscribe to the chrome.storage.onChange event 
But I don't know how in PopUp view force Zustand to reload state from the persistent storage.

Thanks in advance,

**Comment by rzfzr** [comment]
Can't you just call rehydration on the popup page? BTW, are you using local storage? I'm trying to use chrome.storage.sync
